src/main/java/org/fogbowcloud/app/core/models/command/Command.java

```diff
@@ -0,0 +1,106 @@
+package org.fogbowcloud.app.core.models.command;
+
+import javax.persistence.*;
+import javax.validation.constraints.Size;
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * A command is a unit of a task. Each line of a task is a command.
+ */
+@Entity
+@Table(name = "command")
+public class Command implements Serializable {
+
+    private static final transient long serialVersionUID = 5281647552435522413L;
+
+    private static final int UNDETERMINED_RESULT = Integer.MAX_VALUE;
+
+    private static final String RAW_COMMAND_COLUMN_NAME = "raw_command";
+    private static final String EXIT_CODE_COLUMN_NAME = "exit_code";
+    private static final int COMMAND_MAX_SIZE = 2048;
+    private static final int RAW_COMMAND_MAX_SIZE = 1024;
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
+
+    @Column
+    @Size(max = COMMAND_MAX_SIZE)
+    private String command;
+
+    @Column(name = RAW_COMMAND_COLUMN_NAME)
+    @Size(max = RAW_COMMAND_MAX_SIZE)
+    private String rawCommand;
+
+    @Enumerated(EnumType.STRING)
+    private CommandState state;
+
+    @Column(name = EXIT_CODE_COLUMN_NAME)
+    private int exitCode;
+
+    public Command() {}
+
+    public Command(String command, CommandState state) {
```

Comment:
Shouldn't this builder be private?

Reply:
Fixed in 9beb9b63b1b32e6bcb8c2bd1fcdfb05a6d03a48f.